tests/config/test_themes.py

```diff
@@ -19,12 +21,75 @@
     "zt_blue",
     "gruvbox_dark24",
 }
+aliases_16_color = [
+    "default",
+    "black",
+    "dark red",
+    "dark green",
+    "brown",
+    "dark blue",
+    "dark magenta",
+    "dark cyan",
+    "dark gray",
+    "light red",
+    "light green",
+    "yellow",
+    "light blue",
+    "light magenta",
+    "light cyan",
+    "light gray",
+    "white",
+]
 
 
 def test_all_themes():
     assert all_themes() == list(THEMES.keys())
 
 
+# Check built-in themes are complete for quality-control purposes
+@pytest.mark.parametrize(
+    "theme_name",
+    [
+        theme
+        if theme in expected_complete_themes
+        else pytest.param(theme, marks=pytest.mark.xfail(reason="incomplete"))
+        for theme in NEW_THEMES
+    ],
+)
+def test_new_builtin_theme_completeness(theme_name):
+    theme = NEW_THEMES[theme_name]
+    theme_styles = theme.STYLES
+    theme_colors = theme.Color
+
+    # Check if STYLE and REQUIRED_STYLES use the same styles.
+    assert len(theme_styles) == len(REQUIRED_STYLES)
+    assert all(required_style in theme_styles for required_style in REQUIRED_STYLES)
+    # Check if colors are defined with all 3 color codes.
+    for color in theme_colors:
+        if "__" in color.name:
+            continue
+
+        codes = color.value.split()
+        assert len(codes) == 3
+        # Check if 16-color alias is correct
+        assert codes[0].replace("_", " ") in aliases_16_color
+        # Check if 24-bit color is #000000-#ffffff
+        assert re.match("(?:#[\\da-f]{6}|default)$", codes[2])
+        # Check if 256 color is #000-#fff or h0-h255 or g0-g100 0r g#00-g#ff
+        code_256 = re.match(
+            "#[\\da-f]{3}|(?:h|g)([\\d]{1,3})|g#[\\da-f]{2}|default$", codes[1]
+        )
+        assert code_256
+        if code_256.group(1) and code_256.group(0).startswith("h"):
+            assert int(code_256.group(1)) < 256
+        elif code_256.group(1) and code_256.group(0).startswith("g"):
+            assert int(code_256.group(1)) <= 100
```

Comment:
I appreciate the idea here, but I'm not sure if the 256 and 24-bit colors are separate - ie. can the 256 color names be used in 24-bit depth?
@rht Do you know anything about this?

Reply:
Yeah, technically you could use 256 colors and operate them in 24bit mode or use 24bit colors and operate them in 256 color mode where urwid takes care of the conversion internally, but whats the use? We want to use 24bit colors in 24 depth and 256 colors in 256 depth and also down conversion from 24 to 256 is approximate and not good.
I think its easier to search for proper colors online than to rely on conversion.

Reply:
#fff == #ffffff, so some are definitely valid across color depths, and do not lose information. I think we'll want to try and be forgiving with user theme color specification.

That said, these are tests for built-in themes only, so I suppose that's fine for now.